en/references/tested-judge/dsl/README.md

```diff
@@ -0,0 +1,965 @@
+```
+
+#### literal block
+YAML support multiple bloknotations, whereby newlines are kept.
+Hereby will the indentation, that deviates from the alignment not be trimmed away.
+There will be also no escaping.
+
+##### Default (`|`)
+Removes each empty line at the end of the string, but keeps one newline character at the end.
+Example YAML:
+```yaml
+literal: |
+  line1a\nline1b
+    line2
+  # not a comment
+  end
+
+# end of statement
+```
+
+Translation JSON:
+```json
+{"literal": "line1a\\nline1b\n  line2\n# not a comment\nend\n"}
+```
+
+##### Keep (`|+`)
+Keeps all newlines at the end of the string.
+Example YAML:
+```yaml
+literal: |+
+  line1a\nline1b
+    line2
+  # not a comment
+  end
+
+# end of statement
+```
+
+Translation JSON:
+```json
+{"literal": "line1a\\nline1b\n  line2\n# not a comment\nend\n\n"}
+```
+
+##### Trim (`|-`)
+Hereby will each newline at the end be removed and there will be no newline at the end of the string.
+Example YAML:
+```yaml
+literal: |-
+  line1a\nline1b
+    line2
+  # not a comment
+  end
+
+# end of statement
+```
+
+Translation JSON:
+```json
+{"literal": "line1a\\nline1b\n  line2\n# not a comment\nend"}
+```
+
+#### Folded block
+YAML also support multiple block notations whereby newline are replaced by a space, when te next line the alignment respects.
+Each empty line will be interpreted as a newline.
+Also will an alignment, that deviates not be trimmed away.
+There will also be no escaping.
+
+##### Default (`>`)
+Removes each empty line at the end of the string, but keeps one newline character at the end.
+Example YAML:
+```yaml
+literal: >
+  line1a
+  line1b
+    line2
+
+  line3
+
+# end of statement
+```
+
+Translation JSON:
+```json
+{"literal": "line1a line1b\n  line2\n\nline3\n"}
+```
+
+##### Keep (`>+`)
+Keeps all newlines at the end of the string.
+Example YAML:
+```yaml
+literal: >+
+  line1a
+  line1b
+    line2
+
+  line3
+
+# end of statement
+```
+
+Translation JSON:
+```json
+{"literal": "line1a line1b\n  line2\n\nline3\n\n"}
+```
+
+##### Trim (`>-`)
+Removes each empty line at the end of the string and there will be no newline character at the end.
+Example YAML:
+```yaml
+literal: >-
+  line1a
+  line1b
+    line2
+
+  line3
+
+# end of statement
+```
+
+Translation JSON:
+```json
+{"literal": "line1a line1b\n  line2\n\nline3"}
+```
+
+### 'yes' and 'no'
+`yes` and `no` are an extra notation for the booleans `true` and `false`.
+
+### Translation to textual types
+If no strings are used for `stdin`,` stdout` and `stderr`, but numbers and boolean values, the final textual data may deviate from what you intended.
+Example:
+```yaml
+stdin: 077
+```
+
+Will be interpreted as an octal number: `7 ⋅ 8 + 7 = 63`.
+```yaml
+stdin: "63"
+```
+
+### Function name
+De style convention for function names in the DSL is `snake_case`.
+When you deviate from this convention, it can't be guaranteed that the correct style convention for a specific programming languages could be followed.
```

Comment:
```suggestion
When you deviate from this convention, it can't be guaranteed that the correct style convention for a specific programming languages will be used.
```